### PR TITLE
linkerd: init at 2019-07-26

### PR DESCRIPTION
**Pull request by Gonzih**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Gonzih** [comment]
@yurrriq you might be also interested in this one

**Comment by Gonzih** [comment]
@GrahamcOfBorg build kubernetes-linkerd

**Comment by yurrriq** [comment]
Should we just call this `linkerd`?

File: pkgs/applications/networking/cluster/linkerd/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "kubernetes-linkerd-${version}";
+  version = "stable-2.4.0";
+
+  src = fetchurl {
+    url = "https://github.com/linkerd/linkerd2/releases/download/${version}/linkerd2-cli-${version}-linux";
```
Comment (yurrriq) [inline pkgs/applications/networking/cluster/linkerd/default.nix]: Why not build from source instead?

Comment (Gonzih) [reply pkgs/applications/networking/cluster/linkerd/default.nix]: I tried, but failed miserably, was getting module error (unable to detect module path or something like that). Tried to dig through linkerd build scripts, but they are just too convoluted, so gave up and just pulled in binary for now.

**Comment by Gonzih** [comment]
@yurrriq here is my test package that uses buildGoModule

```nix
with import <nixpkgs> {};

buildGoModule rec {
  pname = "linkerd";
  version = "stable-2.4.0";

  src = fetchFromGitHub {
    owner = "linkerd";
    repo = "linkerd2";
    rev = version;
    sha256 = "1r08i7sa45msir14qzjqibhlg01shjsq376awn0mw73mkpgzr859";
  };

  postBuild = "GO111MODULE=on go generate -mod=readonly ./cli";

  modSha256 = "1q5g900000000001111111111111111111y333333m6kjvhlcn87";

  subPackages = [ "cli/cmd" ];

}
```

**Comment by Gonzih** [comment]
Figured out how to build from source, seems like only 2 days ago this particular project was switched to go modules.

File: pkgs/applications/networking/cluster/linkerd/default.nix (hunk 1 above)
Comment (Gonzih) [reply pkgs/applications/networking/cluster/linkerd/default.nix]: fixed

File: pkgs/applications/networking/cluster/linkerd/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub, buildGoModule }:
+
+buildGoModule rec {
+  pname = "linkerd";
+  version = "065dd3ec9d4b84c90383b31be16ccadd34a8ab0d";
```
Comment (kalbasit) [inline pkgs/applications/networking/cluster/linkerd/default.nix]: The version should be the date of the commit formatted as `yyyy-mm-dd`. Also the `pname` should be suffixed with `unstable` if you're not using a released version.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/cluster/linkerd/default.nix (hunk 2 above)
Comment (Gonzih) [reply pkgs/applications/networking/cluster/linkerd/default.nix]: Thanks! Fixed!

**Comment by kalbasit** [comment]
@GrahamcOfBorg build linkerd

**Review by kalbasit — APPROVED** [review]

**Comment by Gonzih** [comment]
Thank you!

**Pull request merged**